fabfile.py

```diff
@@ -13,13 +13,41 @@
 CURRENT_DIR = os.path.dirname(__file__)
 
 
+TIERS = {
+    'production': {
+        'hosts': ['root@104.131.92.59'],
+    },
+    'staging': {
+        'hosts': ['root@162.243.226.226'],
+    },
+}
+
+
+def tier_set(tier_name='staging'):
+    env.tier = tier_name
+    for option, value in TIERS[env.tier].items():
+        setattr(env, option, value)
+
+
+@task
+def production():
+    tier_set('production')
+
+
+@task
+def staging():
+    tier_set('staging')
+
+
 @task
 def install_salt():
+    require('tier', provided_by=(staging,production))
```

Comment:
D'oh, shoulda run flake8.

What this line does is require that the tier variable be set. The `provided_by` flag I think is just to tell the user what they can do to set it if it's not, e.g. this help message:

```
$ fab -H some.website.com deploy
[some.website.com] Executing task 'deploy'

Fatal error: The command 'deploy' failed because the following required environment variable was not defined:
    tier

Try running one of the following commands prior to this one, to fix the problem:
    staging
    production

Aborting.
```